creator/data_reviews/mutations.py

```diff
@@ -0,0 +1,323 @@
+from django_fsm import TransitionNotAllowed
+import graphene
+from graphql import GraphQLError
+from graphql_relay import from_global_id
+from django.db import transaction
+
+from creator.data_reviews.nodes import DataReviewNode
+from creator.studies.models import Study
+from creator.data_reviews.models import DataReview, State
+
+
+def check_review_files(input, study):
+    """
+    Check if files exist and they are from same study
+    """
+    review_file_ids = []
+    if input.get("files"):
+        study_file_ids = {f.pk for f in study.files.all()}
```

Comment:
Maybe premature optimization, but this requires getting all files for the study to validate input files.
You could reduce the amount of data queried by just requesting the kf_ids and comparing those:
```study.files.values("kf_id", flat=True)```
Or you could use the overlap operation:
```
if study.files.filter(kf_id__overlap=input['files']).count() < len(input['files']):
    raise ...
```